File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = 0
        dct = {}
        for i in range(len(nums)):
            d = target - nums[i]
            if d in dct:
                return [i, dct[d]]
            dct[nums[i]] = i

File: 0001-two-sum/0001-two-sum.java
class Solution {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] res = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (map.containsKey(diff)) {
                res[0] = i;
                res[1] = map.get(diff);
            } else {
                map.putIfAbsent(nums[i], i);
            }
        }

        return res;
    }
}
